config/src/config/consensus_config.rs

```diff
@@ -52,6 +53,7 @@ impl Default for ConsensusConfig {
             quorum_store_pull_timeout_ms: 1000,
             quorum_store_poll_count: 20,
             intra_consensus_channel_buffer_size: 10,
+            max_failed_authors_to_store: 10,
```

Comment:
For each block, we store failed authors from the rounds after the previous successful block.
so if successful rounds are 0, 1, 4, 6, 7, we will have 2 failed_authors for block in round 4, and 1 failed_author for block in round 6. 

Window we use in production is over how many blocks are we going to aggregate those. 

If we have 50% block failure rate in production, limit of 10 here is going to be hit in 0.1% of blocks only (0.5^10). So limit of 10 here is plenty, and will rarely be hit.

Reply:
@ikabiljo - Thanks for the clarification. What will happen if network is down for an extended period? Will we have only one failed block in that case or multiple failed blocks? In case it is multiple failed blocks, then I guess the limit of 10 will not suffice?

Reply:
@zekun000 - I thought this was on-chain config (as proposer_type is there for example, which definitely needs to be the same for all nodes).

If this is not on-chain config - where is the on-chain config?



What we want to use here, is penalize nodes that are bad. If we have outage of the network, and have more than 10 failed blocks, it doesn't really matter that we will not penalize all of them - it is probably not their fault anyways.
so limit there is 10 to keep enough useful information, without potentially inflating the block size when network is having issues (i.e. without limit, it would be harder to recover, as block would grow more as outage lasts more)

Reply:
chatted with @zekun000 - there are more fields in local config that we should move to on-chain config. 
So I will leave it as is in this diff, and move all those fields to on-chain config in a separate one.